targettests/Remote-Sim/state_init.cpp

```diff
@@ -0,0 +1,84 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+// REQUIRES: remote-sim
+// REQUIRES: c++20
+
+// clang-format off
+// RUN: nvq++ %cpp_std --enable-mlir --target remote-mqpu -fkernel-exec-kind=2 %s -o %t && %t | FileCheck %s
```

Comment:
I haven't been tracking all the new stuff regarding kernel launches. Will a user be expected to know when to do `-fkernel-exec-kind=2`? If so, it should probably be documented in the user-facing docs, and if not, all the spots that need edits should probably be tracked with some sort of `TODO-FIX-KERNEL-EXEC` (or something like that).

Reply:
This should be temporary and `-fkernel-exec-kind=2` will be removed after we switch to the new launch by default. Will add the markings and an item in https://github.com/NVIDIA/cuda-quantum/issues/1931

Reply:
Added `// TODO-FIX-KERNEL-EXEC` in all tests that use the new launch and added an item in #1931